cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -74,6 +74,46 @@ static auto execute_callback_with_typed_array(SimpleArrayPlex & arrayplex, C &&
 #undef DECL_MM_RUN_CALLBACK_WITH_TYPED_ARRAY
 }
 
+/// Execute the callback function with the typed array
+/// @tparam C the type of the callback function
+/// @param arrayplex the plex array, which is the wrapper of the typed array
+/// @param callback the callback function, which has the immutable typed array as the argument
+/// @return the return type of the callback function
+template <typename C>
+// NOLINTNEXTLINE(misc-use-anonymous-namespace)
+static auto execute_callback_with_typed_array(const SimpleArrayPlex & arrayplex, C && callback)
+{
+// We get the typed array from the arrayplex and call the callback function with the typed array.
+#define DECL_MM_RUN_CALLBACK_WITH_CONST_TYPED_ARRAY(DataType, ArrayType)                \
```

Comment:
Can we merge the `const` and non-`const` versions of the macro to one?

Reply:
I successfully figured out how to make it.

Reply:
This is nice